lib/puppet/util/package/version/pip.rb

```diff
@@ -0,0 +1,230 @@
+module Puppet::Util::Package::Version
+  class Pip
+    def initialize(version)
+      matched = version.match(Regexp.new(("^\\s*") + @@VERSION_PATTERN + ("\\s*$"), Regexp::EXTENDED | Regexp::MULTILINE | Regexp::IGNORECASE))
+
+      raise ValidationFailure.new(version) unless matched
+      self.class.include Comparable
+
+      @epoch_data   = matched[:epoch].to_i
+      @release_data = matched[:release].split('.').map{ |r| r.to_i }                                 if matched[:release]
+      @pre_data     = parse_letter_version(matched[:pre_l], matched[:pre_n])                         if matched[:pre_l]  || matched[:pre_n]
+      @post_data    = parse_letter_version(matched[:post_l], matched[:post_n1] || matched[:post_n2]) if matched[:post_l] || matched[:post_n1] || matched[:post_n2]
+      @dev_data     = parse_letter_version(matched[:dev_l], matched[:dev_n])                         if matched[:dev_l]  || matched[:dev_n]
+      @local_data   = parse_local_version(matched[:local])                                           if matched[:local]
+
+      @key = compose_key(@epoch_data, @release_data, @pre_data, @post_data, @dev_data, @local_data)
+    end
+
+    def self.parse(version)
+      new(version)
+    end
+
+    def self.compare(version_a, version_b)
+      version_a = new(version_a) unless version_a.is_a?(self.class)
+      version_b = new(version_b) unless version_a.is_a?(self.class)
+
+      version_a <=> version_b
+    end
+
+    def to_s
```

Comment:
AptVersion also defines an alias for `to_s` as inspect. Should  Pip also have it, for consistency